workspace/resource_notebook.go

```diff
@@ -431,7 +442,13 @@ func ResourceNotebook() *schema.Resource {
 			})
 		},
 		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			return NewNotebooksAPI(ctx, c).Delete(d.Id(), true)
+			objType := d.Get("object_type")
+			return NewNotebooksAPI(ctx, c).Delete(d.Id(), !(objType == Notebook || objType == File))
 		},
 	}.ToResource()
 }
+
+func isParentDoesntExistError(err error) bool {
+	errStr := err.Error()
+	return strings.HasPrefix(errStr, "The parent folder ") && strings.HasSuffix(errStr, " does not exist.")
```

Comment:
Do we not have an error code for this? Note: once the next Go SDK is released, we can do `errors.is(err, databricks.ErrResourceDoesNotExist)` to check for this case (merged today)!

Reply:
No, we don't have error code at this level.